kwave/kspaceFirstOrder3D.py

```diff
@@ -298,6 +298,13 @@ def kspaceFirstOrder3D(
     # start the timer and store the start time
     TicToc.tic()
 
+    # Currently we only support binary execution, meaning all simulations must be saved to disk.
+    if not simulation_options.save_to_disk:
+        if execution_options.is_gpu_simulation:
+            raise ValueError('GPU simulation requires saving to disk. Please set SimulationOptions.save_to_disk=True')
+        else:
+            raise ValueError('CPU simulation requires saving to disk. Please set SimulationOptions.save_to_disk=True')
+
```

Comment:
Repeated code, perhaps move to a utility function?

Reply:
I already have an issue open to refactor the 2D3D functions due to a lot of repeated code. Next point release?